src/ripple/conditions/condition.h

```diff
@@ -0,0 +1,118 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_CONDITIONS_CONDITION_H
+#define RIPPLE_CONDITIONS_CONDITION_H
+
+#include <ripple/conditions/impl/base64.h> // use Beast implementation
+#include <ripple/conditions/impl/utils.h>
+#include <boost/optional.hpp>
+#include <array>
+#include <cstdint>
+#include <string>
+
+namespace ripple {
+namespace cryptoconditions {
+
+// NIKB-TODO: These should move to a separate file:
+// NIKB-TODO: These should be `enum class : std::uint16_t`
+std::uint16_t constexpr condition_hashlock         = 0;
+std::uint16_t constexpr condition_prefix_sha256    = 1;
+std::uint16_t constexpr condition_threshold_sha256 = 2;
+std::uint16_t constexpr condition_rsa_sha256       = 3;
+std::uint16_t constexpr condition_ed25519          = 4;
+
+// NIKB-TODO: These should be `enum class : std::uint32_t`
+std::uint32_t constexpr feature_sha256             = 1;
+std::uint32_t constexpr feature_preimage           = 2;
+std::uint32_t constexpr feature_prefix             = 4;
+std::uint32_t constexpr feature_threshold          = 8;
+std::uint32_t constexpr feature_rsa_pss            = 16;
+std::uint32_t constexpr feature_ed25519            = 32;
+
+struct condition_t
+{
+    std::uint16_t type;
+    std::uint16_t fulfillment_length;
```

Comment:
I thought the field was the maximum length of a fullfillment. I.e. "reject any fullfillment greater than this", not "reject any fulfillment not equal to this". Did I misunderstand the point of this member?